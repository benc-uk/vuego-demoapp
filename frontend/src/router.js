//
// App router
// ----------------------------------------------
// Ben C, April 2018, Updated for Vue3 2021
//

import Home from './components/Home.vue'
import About from './components/About.vue'
import Error from './components/Error.vue'
import Info from './components/Info.vue'
import Monitor from './components/Monitor.vue'
import Weather from './components/Weather.vue'
import User from './components/User.vue'
import { createRouter, createWebHashHistory } from 'vue-router'

const router = createRouter({
  history: createWebHashHistory(),
  routes: [
    {
      path: '/',
      name: 'home',
      component: Home
    },
    {
      path: '/home',
      name: 'apphome',
      component: Home
    },
    {
      path: '/info',
      name: 'info',
      component: Info
    },
    {
      path: '/monitor',
      name: 'monitor',
      component: Monitor
    },
    {
      path: '/weather',
      name: 'weather',
      component: Weather
    },
    {
      path: '/about',
      name: 'about',
      component: About
    },
    {
      path: '/user',
      name: 'user',
      component: User
    },
    {
      path: '/:catchAll(.*)',
      name: 'catchall',
      component: Error
    }
  ]
})

export default router
