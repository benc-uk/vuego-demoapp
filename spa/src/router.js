//
// App router
// ----------------------------------------------
// Ben C, April 2018
//

import Vue from 'vue'
import Router from 'vue-router'
import Home from './components/Home.vue'
import About from './components/About.vue'
import Error from './components/Error.vue'
import Info from './components/Info.vue'
import Monitor from './components/Monitor.vue'
import Weather from './components/Weather.vue'
import User from './components/User.vue'

Vue.use(Router)

export default new Router({
  mode: 'history',
  routes: [
    {
      path: '/',
      name: 'home',
      component: Home
    },
    {
      path: '/home',
      name: 'apphome',
      component: Home
    },
    {
      path: '/info',
      name: 'info',
      component: Info
    },
    {
      path: '/monitor',
      name: 'monitor',
      component: Monitor
    },
    {
      path: '/weather',
      name: 'weather',
      component: Weather
    },
    {
      path: '/about',
      name: 'about',
      component: About
    },
    {
      path: '/user',
      name: 'user',
      component: User
    },
    {
      path: '*',
      name: 'catchall',
      component: Error
    }
  ]
})
