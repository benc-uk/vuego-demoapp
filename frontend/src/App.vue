<!--
Core app component
----------------------------------------------
// Ben C, April 2018, Updated for Vue3 2021
-->

<template>
  <div>
    <nav class="navbar navbar-expand-lg navbar-dark bg-primary">
      <div class="container-fluid">
        <a class="navbar-brand text-light" href="#">Vue &amp; Go Demo</a>
        <button
          class="navbar-toggler"
          type="button"
          data-bs-toggle="collapse"
          data-bs-target="#navbarSupportedContent"
          aria-controls="navbarSupportedContent"
          aria-expanded="false"
          aria-label="Toggle navigation"
        >
          <span class="navbar-toggler-icon"></span>
        </button>
        <div id="navbarSupportedContent" class="collapse navbar-collapse">
          <ul class="navbar-nav me-auto mb-2 mb-lg-0">
            <li class="nav-item">
              <router-link class="nav-link active" to="/home"> <i class="fas fa-home fa-fw"></i> Home</router-link>
            </li>
            <li class="nav-item">
              <router-link class="nav-link active" to="/info"> <i class="fas fa-cog fa-fw"></i> Info</router-link>
            </li>
            <li class="nav-item">
              <router-link class="nav-link active" to="/monitor"> <i class="fas fa-tachometer-alt fa-fw"></i> Monitor</router-link>
            </li>
            <li v-if="weatherEnabled" class="nav-item">
              <router-link class="nav-link active" to="/weather"> <i class="fas fa-umbrella fa-fw"></i> Weather</router-link>
            </li>
            <li class="nav-item">
              <router-link class="nav-link active" to="/about"> <i class="fas fa-info-circle fa-fw"></i> About</router-link>
            </li>
          </ul>
          <div v-if="authEnabled" class="d-flex">
            <button v-if="!user" class="btn btn-info btn-lg" variant="light" @click="doLogin()">
              <i class="fas fa-sign-in-alt"></i>&nbsp; Login
            </button>
            <router-link v-if="user" class="btn btn-info" to="/user" variant="light"> <i class="fas fa-user"></i>&nbsp; User </router-link>
          </div>
        </div>
      </div>
    </nav>

    <div class="container mt-5">
      <router-view v-slot="{ Component }">
        <transition name="route" mode="out-in">
          <component :is="Component" />
        </transition>
      </router-view>
    </div>
  </div>
</template>

<script>
import auth from './services/auth'
import router from './router'
import { config } from './main'

export default {
  data: function () {
    return {
      user: null,
      // We switch the user sign-in feature on or off depending if auth has been configured
      authEnabled: auth.isConfigured(),
      weatherEnabled: config.weatherEnabled
    }
  },

  async created() {
    // Restore any cached or saved local user
    if (auth.isConfigured()) {
      this.user = auth.user()
    }
  },

  methods: {
    async doLogin() {
      try {
        await auth.login()
        this.user = auth.user()
        router.push('user')
      } catch (err) {
        this.error = err.toString()
      }
    },

    clearUser() {
      this.user = null
    }
  }
}
</script>

<style>
html {
  min-height: 100%;
}

.route-enter-from {
  opacity: 0;
  transform: translateX(100px);
}
.route-enter-active {
  transition: all 0.3s ease-out;
}
.route-leave-to {
  opacity: 0;
  transform: translateX(-100px);
}
.route-leave-active {
  transition: all 0.3s ease-in;
}
</style>
