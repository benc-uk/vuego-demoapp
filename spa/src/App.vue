<!--
Core app component, just template no code/logic
----------------------------------------------
Ben C, April 2018
-->

<template>
  <div id="app">
    <b-navbar toggleable="md" type="dark" variant="primary">
      <b-navbar-toggle target="nav_collapse" />
      <b-navbar-brand to="/"> <img src="./assets/vuejs.svg" height="50px" />Vue.js &amp; Go </b-navbar-brand>
      <b-collapse id="nav_collapse" is-nav>
        <b-navbar-nav>
          <b-button to="/home" variant="dark"> <fa icon="home" />&nbsp; Home </b-button>
          <b-button to="/info" variant="dark"> <fa icon="cogs" />&nbsp; Info </b-button>
          <b-button to="/monitor" variant="dark"> <fa icon="tachometer-alt" />&nbsp; Monitor </b-button>
          <b-button to="/weather" variant="dark"> <fa icon="umbrella" />&nbsp; Weather </b-button>
          <b-button to="/about" variant="dark"> <fa icon="info-circle" />&nbsp; About </b-button>
        </b-navbar-nav>

        <b-navbar-nav v-if="authEnabled" class="ml-auto">
          <b-button v-if="!user" variant="light" @click="doLogin()"> <fa icon="sign-in-alt" />&nbsp; Login </b-button>
          <b-button v-if="user" to="/user" variant="light"> <fa icon="user" />&nbsp; User </b-button>
        </b-navbar-nav>
      </b-collapse>
    </b-navbar>

    <br />

    <b-container>
      <transition name="slidedown">
        <router-view @logoutEvent="clearUser()" />
      </transition>
    </b-container>
  </div>
</template>

<script>
import auth from './services/auth'
import router from './router'

export default {
  data: function() {
    return {
      user: {},
      authEnabled: auth.isConfigured()
    }
  },

  async created() {
    // Restore any cached or saved local user
    if (auth.isConfigured()) {
      this.user = auth.user()
    }
  },

  methods: {
    async doLogin() {
      try {
        await auth.login()
        this.user = auth.user()
        router.push('user')
      } catch (err) {
        this.error = err.toString()
      }
    },

    clearUser() {
      this.user = null
    }
  }
}
</script>

<style>
html {
  font-size: 20px;
}
#app,
html,
body {
  height: 100%;
  background-color: #317256;
}
.b-navbar-nav {
  color: red;
}
.slidedown-enter-active {
  transition: transform 0.4s ease-out;
  transform-origin: top;
}
.slidedown-enter {
  transform: scaleY(0);
}
</style>

<style scoped>
.btn {
  margin-right: 0.3rem;
  width: 7rem;
}
</style>
