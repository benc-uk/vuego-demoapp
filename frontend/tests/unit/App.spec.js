import { mount } from '@vue/test-utils'
import App from '@/App.vue'
import router from '@/router'

describe('App.vue', () => {
  it('renders main app screen', async () => {
    const wrapper = mount(App, {
      global: {
        plugins: [router]
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders navigates to about', async () => {
    const wrapper = mount(App, {
      global: {
        plugins: [router]
      }
    })

    router.push('/about')
    await router.isReady()
    expect(wrapper.html()).toMatchSnapshot()
  })
})
