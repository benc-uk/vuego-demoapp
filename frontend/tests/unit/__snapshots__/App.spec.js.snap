// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.vue renders main app screen 1`] = `
<div>
  <nav class="navbar navbar-expand-lg navbar-dark bg-primary">
    <div class="container-fluid"><a class="navbar-brand text-light" href="#">Vue &amp; Go Demo</a><button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation"><span class="navbar-toggler-icon"></span></button>
      <div id="navbarSupportedContent" class="collapse navbar-collapse">
        <ul class="navbar-nav me-auto mb-2 mb-lg-0">
          <li class="nav-item"><a href="#/home" class="nav-link active"><i class="fas fa-home fa-fw"></i> Home</a></li>
          <li class="nav-item"><a href="#/info" class="nav-link active"><i class="fas fa-cog fa-fw"></i> Info</a></li>
          <li class="nav-item"><a href="#/monitor" class="nav-link active"><i class="fas fa-tachometer-alt fa-fw"></i> Monitor</a></li>
          <!--v-if-->
          <li class="nav-item"><a href="#/about" class="nav-link active"><i class="fas fa-info-circle fa-fw"></i> About</a></li>
        </ul>
        <!--v-if-->
      </div>
    </div>
  </nav>
  <div class="container mt-5">
    <transition-stub>
      <!---->
    </transition-stub>
  </div>
</div>
`;

exports[`App.vue renders navigates to about 1`] = `
<div>
  <nav class="navbar navbar-expand-lg navbar-dark bg-primary">
    <div class="container-fluid"><a class="navbar-brand text-light" href="#">Vue &amp; Go Demo</a><button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation"><span class="navbar-toggler-icon"></span></button>
      <div id="navbarSupportedContent" class="collapse navbar-collapse">
        <ul class="navbar-nav me-auto mb-2 mb-lg-0">
          <li class="nav-item"><a href="#/home" class="nav-link active"><i class="fas fa-home fa-fw"></i> Home</a></li>
          <li class="nav-item"><a href="#/info" class="nav-link active"><i class="fas fa-cog fa-fw"></i> Info</a></li>
          <li class="nav-item"><a href="#/monitor" class="nav-link active"><i class="fas fa-tachometer-alt fa-fw"></i> Monitor</a></li>
          <!--v-if-->
          <li class="nav-item"><a href="#/about" class="router-link-active router-link-exact-active nav-link active" aria-current="page"><i class="fas fa-info-circle fa-fw"></i> About</a></li>
        </ul>
        <!--v-if-->
      </div>
    </div>
  </nav>
  <div class="container mt-5">
    <transition-stub>
      <div class="card border-light mb-3 rounded-3">
        <h4 class="card-header bg-info"><i class="fas fa-info-circle fa-fw"></i> About</h4>
        <div class="card-body">
          <div>Developed by Ben Coleman, 2018~2021</div>
          <ul class="list-group m-4">
            <li class="list-group-item d-flex justify-content-between align-items-center"> App Version <span class="badge bg-primary rounded-pill p-2">3.0.0</span></li>
            <li class="list-group-item d-flex justify-content-between align-items-center"> API Endpoint <span class="badge bg-primary rounded-pill p-2">/api</span></li>
          </ul>
        </div>
      </div>
    </transition-stub>
  </div>
</div>
`;
